module/state_synchronization/execution_data_storer.go

```diff
@@ -216,7 +182,7 @@ func (bw *BlockWriter) Write(p []byte) (n int, err error) {
 		}
 	}
 	if len(p) > 0 {
-		bw.buf = make([]byte, 0, bw.maxBlockSize)
+		bw.buf = make([]byte, 0, len(p))
```

Comment:
Either one will work. If more data needs to be added to the buffer in a subsequent write, it will just be appended to the array and the array will automatically resize if it needs to.

Using `bw.maxBlockSize` means that if we do have subsequent writes before flushing, we won't need to resize the array.

On the other hand, if we don't have any subsequent writes (ie Flush is called right after this call to Write returns), then we overallocated memory for the array by allocating space for `bw.maxBlockSize` when we only needed `len(p)`.

Reply:
So it just depends how many times Write is called on average before each Flush. If the number is 1, then using `len(p)` is better. Otherwise, using `bw.maxBlockSize` may be better.